repl/src/test/scala/ammonite/repl/ScriptTests.scala

```diff
@@ -299,6 +303,55 @@ object ScriptTests extends TestSuite{
             assert(n == expected)
           }
         }
+
+        'processModuleCaching{
+          def check(script: RelPath){
+            val storage = new Storage.Folder(tempDir)
+            val interp1 = createTestInterp(
+              storage,
+              ammonite.repl.Main.defaultPredefString
+            )
+            interp1.replApi.load.module(resourcesPath/script)
+            assert(interp1.compiler != null)
+            val interp2 = createTestInterp(
+              storage,
+              ammonite.repl.Main.defaultPredefString
+            )
+            assert(interp2.compiler == null)
+            interp2.replApi.load.module(resourcesPath/script)
+            assert(interp2.compiler == null)
+          }
+
+          'testOne - check('scriptLevelCaching/"scriptTwo.scala")
+          'testTwo - check('scriptLevelCaching/"scriptOne.scala")
+          'testThree - check('scriptLevelCaching/"QuickSort.scala")
+          'testLoadModule - check('scriptLevelCaching/"testLoadModule.scala")
+          'testFileImport - check('scriptLevelCaching/"testFileImport.scala")
+
+        }
+
+        'testRunTimeExceptionForCachedScripts{
+          val storage = new Storage.Folder(tempDir)
+          rm(resourcesPath/'scriptLevelCaching/"num.value")
+          write(resourcesPath/'scriptLevelCaching/"num.value", "1")
+          val interp1 = createTestInterp(
+            storage,
+            ammonite.repl.Main.defaultPredefString
+          )
+          interp1.replApi.load.module(resourcesPath/'scriptLevelCaching/"runTimeExceptions.scala")
+          val interp2 = createTestInterp(
+            storage,
+            ammonite.repl.Main.defaultPredefString
+          )
+          val res = intercept[java.lang.ArithmeticException]{
+            interp2.replApi.load.module(
+              resourcesPath/'scriptLevelCaching/"runTimeExceptions.scala"
+            )
+          }
+          assert(interp2.compiler == null &&
+            res.toString == "java.lang.ArithmeticException:/by zero")
+        }
+
```

Comment:
runTimeExceptions.scala reads a number from file num.val and prints `5/num.val`, first it is 1 and then changed to zero by the script, so when next time it reads it causes a run-time exception